simpeg_drivers/components/factories/misfit_factory.py

```diff
@@ -186,14 +197,58 @@ def create_nested_simulation(
         :param padding_cells: Number of padding cells around the local survey.
         """
         survey, indices, ordering = inversion_data.create_survey(
-            mesh=mesh, local_index=indices, channel=channel
+            mesh=global_mesh, local_index=indices, channel=channel
         )
         local_sim, mapping = inversion_data.simulation(
-            mesh,
+            global_mesh,
+            local_mesh,
             active_cells,
             survey,
             tile_id=tile_id,
             padding_cells=padding_cells,
         )
+        inv_type = inversion_data.params.inversion_type
+        if inv_type in ["fem", "tdem"]:
+            compute_em_projections(inversion_data, local_sim)
+        elif ("current" in inv_type or "polarization" in inv_type) and (
+            "2d" not in inv_type or "pseudo" in inv_type
+        ):
+            compute_dc_projections(inversion_data, local_sim, indices)
+        return local_sim, np.hstack(indices), ordering, mapping
+
+
+def compute_em_projections(inversion_data, simulation):
+    """
+    Pre-compute projections for the receivers for efficiency.
+    """
+    rx_locs = inversion_data.entity.vertices
+    projections = {}
+    for comp in "xyz":
+        projections[comp] = simulation.mesh.get_interpolation_matrix(
+            rx_locs, "faces_" + comp[0]
+        )
 
-        return local_sim, indices, ordering, mapping
+    for source in simulation.survey.source_list:
+        for receiver in source.receiver_list:
+            proj = 0.0
+            for ori, comp in zip(receiver.orientation, "xyz", strict=False):
```

Comment:
Should we use `strict=True`?  My understanding is that if you expect the lists to have the same length then `strict=False` hides bugs if they turn up with different lengths..

Reply:
Yea, not sure anymore why I set it to false. Reverting